api-get-object-file.go

```diff
@@ -40,11 +38,12 @@ func (c Client) FGetObject(bucketName, objectName, filePath string, opts GetObje
 }
 
 // FGetEncryptedObject - Decrypt and store an object at filePath.
-func (c Client) FGetEncryptedObject(bucketName, objectName, filePath string, materials encrypt.Materials) error {
-	if materials == nil {
-		return ErrInvalidArgument("Unable to recognize empty encryption properties")
+func (c Client) FGetEncryptedObject(bucketName, objectName, filePath string, key encrypt.Key) error {
+	cipher, err := encrypt.NewCipher(encrypt.DareHmacSha256, key)
```

Comment:
Why change the word Cipher? , we kept materials to be inline with what AWS S3 calls this.  Advantages of calling it materials is that users of aws-sdks have a fair understanding of what is providing by our library.  Cipher is more technically correct in the encryption context.. But since its S3 API we can stick to what AWS SDKs already choose.  For example we should look at aws-sdk-java implementation as the standard for such naming conventions.

Reply:
In general correct. The cipher is actually low-level API (for e.g. GetEncryptedObject we just require a key). So I agree to not bother users with low-level details but I also think we should do it correctly at the low-level. If somebody will use the low-level API he/she (hopefully) knows what he/she is doing. So I think it's okay to use "correct" terms.

AWS "Materials" is a java thing and don't say anything about what is required or what it does...  I don't think it's a good thing to carry bloated java idioms to Go.

Reply:
Okay that is a valid argument. I thought this is a standard representation if it is only aws-sdk-java then we can stick to simpler approach.

Reply:
Also AWS Go sdk has it but you can see there that it's "literally" copied from aws-java...
I think it's okay to do it "correctly" underneath and provide a "nice" high-level API (just the key). From my perspective the AWS materials approach is also not nice from an API point...

Reply:
In that sense we can reference that these are materials from aws-sdk-go . And we choose it to be better as Cipher.

Reply:
Okay